src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -19,7 +22,14 @@ internal sealed class ExceptionProcessor : BaseProcessor<Activity>
     public ExceptionProcessor()
     {
 #if NET6_0_OR_GREATER || NETFRAMEWORK
-        this.fnGetExceptionPointers = Marshal.GetExceptionPointers;
+        if (RuntimeFeature.IsDynamicCodeSupported)
+        {
+            throw new NotSupportedException($"'{typeof(Marshal).FullName}.GetExceptionPointers' is not supported when running native AOT.");
```

Comment:
Is it possible to make this a build time error? I think that will help the developers to discover the issue earlier.